contracts/Joint.sol

```diff
@@ -268,67 +511,68 @@ contract Joint {
         if (balanceOfPair() > 0) masterchef.deposit(pid, balanceOfPair());
     }
 
-    function swapReward(uint256 _rewardBal) internal {
-        // We don't want to sell reward
-        if (ratio == 0) {
-            return;
+    function swapReward(uint256 _rewardBal)
+        internal
+        returns (address _swapTo, uint256 _receivedAmount)
+    {
+        if (reward == tokenA || reward == tokenB || _rewardBal == 0) {
+            return (address(0), 0);
         }
 
-        address swapTo = findSwapTo(reward);
-        //Call swap to get more of the of the swapTo token
-        IUniswapV2Router02(router)
-            .swapExactTokensForTokensSupportingFeeOnTransferTokens(
-            _rewardBal.mul(ratio).div(MAX_RATIO),
-            0,
-            getTokenOutPath(reward, swapTo),
-            address(this),
-            now
-        );
+        _swapTo = findSwapTo(reward);
+        _receivedAmount = sellCapital(reward, _swapTo, _rewardBal);
     }
 
     // If there is a lot of impermanent loss, some capital will need to be sold
     // To make both sides even
     function sellCapital(
         address _tokenFrom,
         address _tokenTo,
-        uint256 _amount
-    ) public onlyGovOrStrategist {
-        IUniswapV2Router02(router)
-            .swapExactTokensForTokensSupportingFeeOnTransferTokens(
-            _amount,
-            0,
-            getTokenOutPath(_tokenFrom, _tokenTo),
-            address(this),
-            now
-        );
+        uint256 _amountIn
+    ) internal returns (uint256 _amountOut) {
+        uint256[] memory amounts =
+            IUniswapV2Router02(router).swapExactTokensForTokens(
+                _amountIn,
+                0,
+                getTokenOutPath(_tokenFrom, _tokenTo),
+                address(this),
+                now
+            );
+        _amountOut = amounts[amounts.length - 1];
     }
 
-    function liquidatePosition() public onlyGuardians {
-        masterchef.withdraw(pid, balanceOfStake());
-        IUniswapV2Router02(router).removeLiquidity(
-            tokenA,
-            tokenB,
-            balanceOfPair(),
-            0,
-            0,
-            address(this),
-            now
-        );
+    function _liquidatePosition() internal returns (uint256, uint256) {
```

Comment:
return `(balanceOfA(), balanceOfB())` to account for balance pre removing liquidity